pkg/server/singleprocess/service_ui_pipeline.go

```diff
@@ -65,7 +65,24 @@ func (s *Service) UI_ListPipelines(
 		}
 		var lastRunBundle *pb.UI_PipelineRunBundle
 		if pipelineLastRun != nil {
-			lastRunBundle = &pb.UI_PipelineRunBundle{PipelineRun: pipelineLastRun}
+			if len(pipelineLastRun.Jobs) != 0 {
+				job, err := s.GetJob(ctx, &pb.GetJobRequest{
+					JobId: pipelineLastRun.Jobs[0].Id,
+				})
+				if err != nil {
+					return nil, hcerr.Externalize(
+						log,
+						err,
+						"failed to get first job for latest pipeline run",
+					)
+				}
+				lastRunBundle = &pb.UI_PipelineRunBundle{
+					PipelineRun: pipelineLastRun,
+					QueueTime:   job.QueueTime,
+					Application: job.Application,
+				}
+			}
```

Comment:
Should we have a warning logged that the server couldn't retrieve any jobs for the last run because the jobs array was empty? Might be nice for debugging in the error case that it is.